From: Joel Winarske <joel.winarske@gmail.com>
Date: Sun, 4 Aug 2024 20:16:41 +0000
Subject: [PATCH] CMake: riscv support

Detect riscv32/riscv64 in CMAKE_SYSTEM_PROCESSOR.

Upstream-Status: Inappropriate

Signed-off-by: Joel Winarske <joel.winarske@gmail.com>
---
  CMakeLists.txt | 6 ++++++
  1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0ef4a68cb7..de0321ac0b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,6 +50,12 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "arm" OR CMAKE_SYSTEM_PROCESSOR MATCHES "aarch
     else()
         set(ARCH "arm")
     endif()
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "riscv32" OR CMAKE_SYSTEM_PROCESSOR MATCHES "riscv64")
+    if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+        set(ARCH "riscv64")
+    else()
+        set(ARCH "riscv32")
+    endif()
 elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^mips.*")
     if(CMAKE_SIZEOF_VOID_P EQUAL 8)
         set(ARCH "mips64el")
